Dockerfile

```diff
@@ -0,0 +1,106 @@
+# Base image: Ubuntu 20.04
+FROM ubuntu:20.04
+LABEL maintainer="CPSWT Team"
+
+# Set it to noninteractive mode
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Install necessary packages
+RUN apt-get update && \
+    apt-get install -y \
+        apt-transport-https \
+        bison \
+        build-essential \
+        ca-certificates \
+        clang \
+        curl \
+        doxygen \
+        flex \
+        gcc \
+        gdb \
+        git \
+        graphviz \
+        libboost1.71-all-dev \
+        libcppunit-dev \
+        libjsoncpp-dev \
+        libosgearth-dev \
+        libqt5opengl5-dev \
+        libwebkit2gtk-4.0-37 \
+        libxml2-dev \
+        lld \
+        make \
+        mongodb \
+        mpi-default-dev \
+        netcat \
+        openjdk-8-jdk \
+        openjdk-17-jdk \
+        openscenegraph-plugin-osgearth \
+        perl \
+        python2 \
+        python3 \
+        python3-pip \
+        python-is-python2 \
+        qt5-qmake \
+        qtbase5-dev \
+        qtbase5-dev-tools \
+        qtchooser \
+        software-properties-common \
+        wget \
+        xterm \
+        zlib1g-dev \
+        unzip \
+        zip
+
+#install Gradle
+ARG GRADLE_VERSION=7.5
```

Comment:
Not sure if this is needed -- the gradle-wrapper (gradle wrapper --gradle-version=8.0) should accomplish that same thing.  You just need to use `./gradlew` instead of `gradle` when executing gradle